app/BookRepository.php

```diff
@@ -0,0 +1,71 @@
+<?php
+
+namespace App;
+
+use App\Models\Book;
+use App\Models\Publisher;
+use App\Models\Writer;
+
+class BookRepository
+{
+    /**
+     * Store a new book with its relations in the database.
+     *
+     * @param array $requestData
+     *
+     * @return \App\Models\Book
+     */
+    public function storeBookWithRelations(array $requestData)
+    {
+        $writer = $this->findOrCreateWriter($requestData);
+        $publisher = $this->findOrCreatePublisher($requestData);
+
+        return Book::create([
+            'title' => $requestData['title'],
+            'ISBN' => $requestData['ISBN'],
+            'publication_year' => $requestData['publication_year'],
+            'price' => $requestData['price'],
+            'genre' => $requestData['genre'],
+            'subgenre' => $requestData['subgenre'],
+            'writer_id' => $writer->id,
+            'publisher_id' => $publisher->id,
+        ]);
+    }
+
+    /**
+     * Find or create a writer.
+     *
+     * @param array $requestData
+     *
+     * @return \App\Models\Writer
+     */
+    protected function findOrCreateWriter(array $requestData)
+    {
+        if ($requestData['writer_id']) {
+            return Writer::find($requestData['writer_id']);
+        }
+
+        return Writer::create([
+            'name' => $requestData['writer_name'],
+        ]);
```

Comment:
Assuming that the acceptance criteria states "if the writer doesn't exist, create a new one", with the current implementation users could enter the same author multiple times since they only enter the `writer_name`.

![writer](https://github.com/AnjaBasara/php-laravel-assignment/assets/25691160/a05a96c4-cd23-4699-aec3-6b10435f2256)

If the acceptance criteria states to enable adding a new Writer through the Create Book form, all the fields from the Create Writer form should be present and those fields should be used in checking if the Writer already exists in the database:

![create writer](https://github.com/AnjaBasara/php-laravel-assignment/assets/25691160/809dfad8-ecf5-4ce5-a7e0-161369daad3a)